src/components/helpers/figure_snap_helper.ts

```diff
@@ -0,0 +1,188 @@
+import { FIGURE_BORDER_WIDTH } from "../../constants";
+import { Figure, Pixel } from "../../types";
+
+const SNAP_MARGIN: Pixel = 5;
+
+type HSnapAxis = "top" | "bottom" | "vCenter";
+type VSnapAxis = "right" | "left" | "hCenter";
+
+export interface SnapLine {
+  matchedFigs: Figure[];
+  position: Pixel;
+  snapOffset: number;
+}
+
+interface SnapReturn {
+  snappedFigure: Figure;
+  verticalSnapLine?: SnapLine;
+  horizontalSnapLine?: SnapLine;
+}
+
+/**
+ * Try to snap the given figure to the other figures when moving the figure, and return the snapped
+ * figure and the possible snap lines, if any were found
+ *
+ * @param figureToSnap figure to snap
+ * @param otherFigures other figure the main figure can snap to
+ */
+export function snapForMove(figureToSnap: Figure, otherFigures: Figure[]): SnapReturn {
+  const snappedFigure = { ...figureToSnap };
+
+  const verticalSnapLine = getSnapLine(snappedFigure, ["left", "right", "hCenter"], otherFigures, [
+    "left",
+    "right",
+    "hCenter",
+  ]);
+
+  const horizontalSnapLine = getSnapLine(
+    snappedFigure,
+    ["top", "bottom", "vCenter"],
+    otherFigures,
+    ["top", "bottom", "vCenter"]
+  );
+
+  snappedFigure.x -= verticalSnapLine?.snapOffset || 0;
+  snappedFigure.y -= horizontalSnapLine?.snapOffset || 0;
+
+  return { snappedFigure, verticalSnapLine, horizontalSnapLine };
+}
+
+/**
+ * Try to snap the given figure to the other figures when resizing the figure, and return the snapped
```

Comment:
Isn't "to the other figures" more english, since thse are the figures given as argument, and not some unspecified figures. Maybe "to the given other figures" but that desn't sounds good.

Reply:
```suggestion
 * Tries to snap the given figure to the other figures when resizing the figure, and return the snapped
```

Reply:
> I guess it won't try to snap to several figures at the same time ?

wrong guess

Reply:
then "to other figures" makes more sense than "to the other figures" imho